### DISCUSSION TITLE
Full scope theme for design tokens and greater theme opportunities

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#521] [Announcements]
## The problem

Auro design tokens have always been a base component of the Auro ecosystem. The working theory has always been that theme support can be achieved by creating an alternate set of design tokens to be injected into the UI. 

The problem as this discussion is concerned is that not all values within a component UI have been tokenized and the values are hard-coded into the element's CSS. This limits the volume of things that can be updated via a new token-based theme and forces the consumer to perform CSS changes if not extend the class of the element to create a custom version. 

## The recommendation 

The recommendation here is for the Auro Design Tokens project to evolve with new and exciting thinking. By adopting concepts like [Component-Specific Design Tokens](https://cloudfour.com/thinks/component-specific-design-tokens/), Auro can greatly expand the scope of theme support allowing teams to easily evolve their UIs based on business and product needs. 

## The expectation 

The expectation is that over the next few sprints, the Auro team, working with design leadership, is expanding the scope of [Auro Design Tokens](https://auro.alaskaair.com/getting-started/developers/design-tokens) to include many more tokens as well additional levels of element scope. 

Here is an [example story for updating auro-button](https://github.com/AlaskaAirlines/auro-button/issues/243).

As elements are updated, releases will be announced.

**Comment by blackfalcon** [comment]
This table represents the status of element that support this new specification of [component-specific design tokens](https://cloudfour.com/thinks/component-specific-design-tokens/).

As of today, all elements support theming, but are limited to the current set of Auro Design Tokens.

| repo | theme support | Issue(s) to follow | Release
|---|---|---|---
|wcss|√||[5.0.0](https://github.com/AlaskaAirlines/WebCoreStyleSheets/releases/tag/v5.0.0)
|icons|√
|design tokens||https://github.com/AlaskaAirlines/AuroDesignTokens/issues/140
|auro-accordion|
|auro-alert||https://github.com/AlaskaAirlines/auro-alert/issues/45 https://github.com/AlaskaAirlines/auro-alert/issues/50
|auro-avatar|
|auro-background|
|auro-backtotop|
|auro-badge|
|auro-banner|
|auro-button||https://github.com/AlaskaAirlines/auro-button/issues/243
|auro-card|
|auro-carousel|
|auro-checkbox||
|auro-combobox||
|auro-datetime|
|auro-datepicker||https://github.com/AlaskaAirlines/auro-datepicker/issues/189 https://github.com/AlaskaAirlines/auro-datepicker/issues/180
|auro-dialog|
|auro-drawer|
|auro-dropdown||
|auro-flight|
|auro-flightline|
|auro-header|
|auro-hyperlink|
|auro-icon|
|auro-input|
|auro-loader|
|auro-lockup|
|auro-menu||
|auro-nav|
|auro-pane||
|auro-popover||
|auro-radio||
|auro-select||
|auro-skeleton|
|auro-sidenav|
|auro-table|
|auro-toast|

**Comment by blackfalcon** [comment]
A new feature that we are interested in exploring is companion content with each component page in the Auro doc site that lists our all the tokens used for a specific element. Think, [this tokens page](https://auro.alaskaair.com/getting-started/developers/design-tokens), but only a subset used for that element. And a clear listing if this is a global token, or level 2 or 3.

**Comment by blackfalcon** [comment]
re: token levels

> Level 1: Global Tokens
At this level, tokens represent core design decisions like color, typography, sizing, and animation. They aren’t tied to a specific use case. They can be used globally or consumed by other tokens.
>
> Level 2: Application-Specific
Application-specific tokens help communicate the intended use of the token. For example, color-blue doesn’t say much about how or when the color should be used. But color-interactive makes sense to use for links or a CTA.
> 
> Level 3: Component-Specific
We can take this idea one step further with component-specific tokens.
Component-specific tokens represent the properties associated with a component.

Thinking practically about where these things may live and be managed, I propose the following. 

### Level 1

These are GLOBAL tokens. Managed via the Auro Design Tokens repo. A consumable for ALL applications. 

### Level 2

These are tokens added to a project at the GLOBAL level (same as CSS), but are specifically designed for this application. These are tokens that reuse the token names from Level 1, but will have values specific to that application. 

### Level 3

Level 3 is tied directly to the 'component' itself, but how? If the concept of these tokens reside within the shadowDOM, there are issues to how we could allow users the ability to customize at that level. 

Asking a user to install a component-specific stylesheet to their global styles will not be embraced. 

Need to think this one through more.

**Reply by blackfalcon** [reply]
For consideration, here is Adobe's design token visualization tool. 
https://opensource.adobe.com/spectrum-tokens/visualizer/?filter=spectrum%2Clight%2Cdesktop&component=checkbox

**Comment by Patrick-Daly-AA** [comment]
After some discussion between @jason-capsule42 & @braven112 The proposed ideas for naming these three levels of tokens would be as follows:

Level 1: Brand Tokens
Level 2: Semantic Tokens
Level 3: Component Tokens

We feel that this will help address clarity when discussing these tokens to those who aren't as fully immersed in Auro Component development.

More thought will need to go into how we want to organize handling Component Tokens (Level 3), so that we can also provide guidance to implementing teams.
